tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,259 @@
+  # We resize CIFAR10 images to match that size
+  image = tf.image.resize(image, [224, 224])
+  image = tf.keras.applications.mobilenet.preprocess_input(image)
+  return image
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  image_features = tf.map_fn(lambda x: _decode_and_preprocess_image(x[0]),
+                             inputs[IMAGE_KEY], dtype=tf.float32)
+
+  outputs[transformed_name(IMAGE_KEY)] = image_features
+  # TODO(b/157064428): Support label transformation for Keras.
+  # Do not apply label transformation as it will result in wrong evaluation.
+  outputs[transformed_name(LABEL_KEY)] = inputs[LABEL_KEY]
+
+  return outputs
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+  """Train the model based on given args.
+
+  Args:
+    fn_args: Holds args used to train the model as name/value pairs.
+  """
+  tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+
+  train_dataset = _input_fn(fn_args.train_files, tf_transform_output,
+                            _TRAIN_BATCH_SIZE)
+  eval_dataset = _input_fn(fn_args.eval_files, tf_transform_output,
+                           _EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model = _build_keras_model()
+
+  steps_per_epoch = _TRAIN_DATA_SIZE / _TRAIN_BATCH_SIZE
+  total_epochs = int(fn_args.train_steps / steps_per_epoch)
+  classifier_epochs = int(total_epochs / 2)
+  finetune_epochs = total_epochs - classifier_epochs
+
+  # Freeze the whole MobileNet backbone and train the top classifer only
+  base_model = model.get_layer('mobilenet_1.00_224')
+  _freeze_model_by_percentage(base_model, 1.0)
+  # We need to recompile the model because layer properties have changed
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=0.0015),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
+
+  model.fit(
+      train_dataset,
+      epochs=classifier_epochs,
+      steps_per_epoch=steps_per_epoch,
+      validation_data=eval_dataset,
+      validation_steps=fn_args.eval_steps
+  )
+
+  # Optional
+  # Unfreeze the top MobileNet layers and finetune the whole model
+  base_model = model.get_layer('mobilenet_1.00_224')
+  _freeze_model_by_percentage(base_model, 0.9)
+  # We need to recompile the model because layer properties have changed
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=0.0005),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
```

Comment:
I feel that it will be helpful for the users to know the model status after each recompile, as they will be able to see how many parameters are frozen. Maybe we can add a logging before each summary, e.g. "Start training the top classifier" and "Start fine-tuning the model" ?

Reply:
sounds good. as long as it is more clear.